core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1584,7 +1584,11 @@ class SparkContext(config: SparkConf) extends Logging {
       path: String, recursive: Boolean, addedOnSubmit: Boolean, isArchive: Boolean = false
     ): Unit = {
     val uri = if (!isArchive) {
-      new Path(path).toUri
+      if (Utils.isAbsoluteURI(path) && path.contains("%")) {
+        new URI(path)
+      } else {
+        new Path(path).toUri
+      }
     } else {
       Utils.resolveURI(path)
```

Comment:
@sarutak, I think we can just repalce `Path` here too (looks like you're fixing this problem across the codebase, right?)

Reply:
Yeah, the behavior of `Path` seems to be a little bit confusable and there may be lots of places where `Path` does not behave  as we expect.
O.K, I'll looking into it.

Reply:
@HyukjinKwon BTW, Just replacing `Path` is my first idea of this PR.
https://github.com/apache/spark/pull/31718/commits/3c40f988de81600a75190751bf9987dfc93d09e9#diff-1e596e9e9bd50e3075847a66f8fecee741289b2b0478f9099c3bf2b680ebaa35R1586

But I changed to reflect [your comment](https://github.com/apache/spark/pull/31718#issuecomment-792225217).
Anyway, I'll open a followup PR.

Reply:
Yeah, I know :). Sorry for a bit of forth and back. If we change it all in the code base, I think it makes more sense to land a correct fix rather then staying safer and conservative with a bandaid fix